include/pipettes/core/uart_message_handler.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <variant>
+
+#include "common/hal/io.hpp"
+#include "common/hal/motor.hpp"
+#include "pipettes/core/communication.hpp"
+#include "pipettes/core/pipette_messages.hpp"
+
+template <typename T, motor_protocol::MotorProtocol Motor>
+requires io::WriterProtocol<T> class MessageHandler {
+  public:
+    explicit MessageHandler(T &writer, Motor &motor,
+                            communication::MessageWriter &message_writer)
+        : writer(writer), motor(motor), message_writer(message_writer) {}
+
+    void handle(const pipette_messages::ReceivedMessage &message) {
```

Comment:
Could we make this function named something different than the ones that are called by the visitor? I understand what's going on here and the compiler handles it fine probably but to me, it's too easy to go overboard with overloads in c++ and we should really restrict ourselves to only having overloads that
- Provide the same semantic functionality
- Have essentially the same contract (constrained by their inputs)
- Basically, have any other differences between them than what is required to transform different inputs

For instance, it's usually pretty bad to have something like
```c++
class Foo {
public:
    bool bar;
    bool gotBar;
    bool getBar() { gotBar = true; return bar;}
    bool getBar() const { return bar;}
}
```
because they don't do the same thing (aside from handling different kinds of inputs). And here, the `handle` defined on line 17 does something completely different than the `handle`s defined on line 23 and down; it's in its own semantic category.